sdk/go/common/workspace/plugins.go

```diff
@@ -1173,9 +1173,28 @@ func GetPluginInfo(kind PluginKind, name string, version *semver.Version) (*Plug
 //  * if found in the pulumi dir's installed plugins, a PluginInfo and path to the executable
 //  * an error in all other cases.
 func getPluginInfoOrPath(
-	kind PluginKind, name string, version *semver.Version, skipMetadata bool) (*PluginInfo, string, error) {
+	kind PluginKind, name string, version *semver.Version, skipMetadata bool, projectPlugins map[string]*PluginInfo) (*PluginInfo, string, error) {
 	var filename string
 
+	if projectPlugins != nil {
+		plugin, ok := projectPlugins[name]
```

Comment:
Not sure we can be as simple as a map keyed by name. For one there are different kinds of plugins and it would be good to have both a "resource-aws" and "analyser-aws". Secondly we need to think how to handle version when using "link". @danielrbradley might have better ideas here on if link should be treated as versionless, or if there's a use case for being able to specify both aws v4  and aws v5 in one project (that sounds reasonable to me).